src/controls/editor-mode/left-sidebar.js

```diff
@@ -59,82 +61,94 @@ const EditorLeftSidebar = () => {
-                  !((type == types.BAR && numericColumns.length > 1 &&
-                    ((renderableValueKeys.length <= 2 && !addUserControls) || addUserControls)) ||
-                    (type == types.MAP && numericColumns.length > 0 && renderableValueKeys.length === 1))}>
-                  {renderToggle(
-                    type === types.MAP ? 'Add Value Controls' : 'Add Benchmark',
-                    addUserControls,
-                    () => userUpdate({ addUserControls: !addUserControls }),
-                    false,
-                    <Tooltip
-                      description={type === types.MAP ?
-                        'Add data value controls on the widget' :
-                        'Benchmark values are unique values used to compare data with.'
-                      }
-                      width='9rem'
-                      arrow={false}
-                      position='right'
-                      classes={{
-                        container: 'mb-0.5',
-                        content: 'overflow-y-visible',
-                      }}
-                    >
-                      <Icons.AlertInformation
-                        size='sm'
-                        color={getTailwindConfigColor('secondary-500')}
-                      />
-                    </Tooltip>
-                  )}
-                </MutedBarrier>
-              }
-              {type === types.BAR &&
-                Object.values(TOP_COLUMN_KEYS).every(elem => JSON.stringify(columns).includes(elem)) &&
-                <MutedBarrier mute={addUserControls}>
-                  {renderToggle('Add Top Categories',
-                    addTopCategories,
-                    () => {
-                      if (addTopCategories) {
-                        update({ propFilters: [] })
-                      }
-                      userUpdate({ addTopCategories: !addTopCategories })
-                    },
-                    false,
-                    <Tooltip
-                      description={'Add top 10 category controls.'}
-                      width='9rem'
-                      arrow={false}
-                      position='right'
-                      classes={{
-                        container: 'mb-0.5',
-                        content: 'overflow-y-visible',
-                      }}
-                    >
-                      <Icons.AlertInformation
-                        size='sm'
-                        color={getTailwindConfigColor('secondary-500')}
-                      />
-                    </Tooltip>
-                  )}
-                </MutedBarrier>
+      {hasDevAccess()
+        && (type === types.STAT
+          || (type === types.BAR &&
+            (Object.values(TOP_COLUMN_KEYS).every(elem => JSON.stringify(columns).includes(elem))
+            || numericColumns.length > 1))
+          || (type == types.MAP && numericColumns.length > 0 && renderableValueKeys.length === 1)
+        )
```

Comment:
consider extracting these conditions as variable or function call (if latter, define the function outside of the component scope to leverage natural memoization)

Reply:
Done!